src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV1.cs

```diff
@@ -207,28 +209,44 @@ public class DownloadPipelineArtifactTaskV1 : PipelineArtifactTaskPluginBaseV1
             }
             else if (buildType == buildTypeSpecific)
             {
-                int pipelineId;
-                if (buildVersionToDownload == buildVersionToDownloadLatest)
-                {
-                    pipelineId = await this.GetpipelineIdAsync(context, buildPipelineDefinition, buildVersionToDownload, projectName, tagsInput);
-                }
-                else if (buildVersionToDownload == buildVersionToDownloadSpecific)
-                {
-                    pipelineId = Int32.Parse(userSpecifiedpipelineId);
-                }
-                else if (buildVersionToDownload == buildVersionToDownloadLatestFromBranch)
+                int? pipelineId = null;
+
+                bool buildTriggeringBool = false;
+                if (bool.TryParse(buildTriggering, out buildTriggeringBool) && buildTriggeringBool)
                 {
-                    pipelineId = await this.GetpipelineIdAsync(context, buildPipelineDefinition, buildVersionToDownload, projectName, tagsInput, branchName);
+                    string triggeringPipeline = context.Variables.GetValueOrDefault("build.triggeredBy.buildId")?.Value;
+
+                    if (!string.IsNullOrEmpty(triggeringPipeline))
+                    {
+                        pipelineId = int.Parse(triggeringPipeline);
+                    }
                 }
-                else
+
+                if (!pipelineId.HasValue)
                 {
-                    throw new InvalidOperationException("Unreachable code!");
+                    if (buildVersionToDownload == buildVersionToDownloadLatest)
+                    {
+                        pipelineId = await this.GetpipelineIdAsync(context, buildPipelineDefinition, buildVersionToDownload, projectName, tagsInput);
+                    }
+                    else if (buildVersionToDownload == buildVersionToDownloadSpecific)
+                    {
+                        pipelineId = Int32.Parse(userSpecifiedpipelineId);
+                    }
+                    else if (buildVersionToDownload == buildVersionToDownloadLatestFromBranch)
+                    {
+                        pipelineId = await this.GetpipelineIdAsync(context, buildPipelineDefinition, buildVersionToDownload, projectName, tagsInput, branchName);
+                    }
+                    else
+                    {
+                        throw new InvalidOperationException("Unreachable code!");
+                    }
                 }
+
                 downloadParameters = new PipelineArtifactDownloadParameters
                 {
                     ProjectRetrievalOptions = BuildArtifactRetrievalOptions.RetrieveByProjectName,
                     ProjectName = projectName,
-                    PipelineId = pipelineId,
+                    PipelineId = pipelineId.GetValueOrDefault(),
```

Comment:
>GetValueOrDefault [](start = 44, length = 17)

We expect to have a value here at this point so I would fail fast if it was null